src/process.cpp

```diff
@@ -424,6 +439,10 @@ std::optional<proc::proc_t> parse(const std::string &file_name) {
         ctx.image_path = parse_env_val(this_env, *image_path);
       }
 
+      if(id) {
+        ctx.id = parse_env_val(this_env, *id);
```

Comment:
It's probably simpler to assign the index as the ID for apps here if they didn't have one on disk, then we can avoid having to separately handle apps by-index and by-ID. We can just always iterate through the app list and search by ID.

When assigning the ID, we should make sure to still count apps with a true ID in the index, otherwise changing an app from index to ID will reshuffle the apps after it and cause the box art to get mixed up in Moonlight.